hugr-core/src/types/type_param.rs

```diff
@@ -294,7 +302,11 @@ impl TypeArg {
         match self {
             TypeArg::Type { ty } => ty.validate(var_decls),
             TypeArg::BoundedNat { .. } | TypeArg::String { .. } => Ok(()),
-            TypeArg::Sequence { elems } => elems.iter().try_for_each(|a| a.validate(var_decls)),
+            TypeArg::List { elems } => {
+                // TODO: Full validation would check that the type of the elements agrees
```

Comment:
Have you made an issue for this?

Reply:
There was a TODO comment to this effect before.